packages/jest-environment-composedb/index.js

```diff
@@ -57,9 +57,25 @@ export default class ComposeEnvironment extends NodeEnvironment {
   }
 
   async teardown() {
-    await this.global.ceramic.close()
-    await this.global.ipfs.stop()
-    await this.tmpFolder.cleanup()
-    await super.teardown()
+    try {
+      await this.global.ipfs.stop()
+    } catch (e) {
+      //ignore
```

Comment:
this worries me.  We should be aiming to maintain the ability to do a clean shutdown, this will just mask if we have a problem there.  Were you actually hitting errors here?

Reply:
Yep, was actually hitting errors here, which sadly can't be handled by this. It actually took reordering of the stop.

Reply:
What are the errors?  Why are we only seeing them here now?  Is there a regression of some kind?  At the very least we should file a ticket and add a TODO here to remove the try/catches once the shutdown issue is fixed, if we can't just go ahead and fix the shutdown issue directly.

Reply:
The main one that occurs is `aborted` on sqlite. As ceramic is in the process of closing, it receives additional events it attempts to write to sqlite (maybe from IPFS?), which then causes an error, and the test fails. Given that this is a test, this feels like the correct behavior, since it shouldn't be up to this library's tests to determine if ipfs and ceramic are closing properly.

Reply:
https://linear.app/3boxlabs/issue/CDB-2615/determine-why-shutting-down-ceramic-before-ipfs-causes-aborted

Reply:
I don't think it's a good idea to merge changes breaking the test setup and have a "let's fix it later" ticket in the backlog.
If the Ceramic RC fails to properly close its processes, that seems like a high priority issue that should be addressed IMO.

Reply:
That should be addressed in js-ceramic, but should not composedb RC's. At worst, this should be a known issues for this RC.